packages/headless/src/features/generated-answer/generated-answer-selectors.ts

```diff
@@ -1,17 +1,23 @@
+import {createSelector} from '@reduxjs/toolkit';
 import {SearchAppState} from '../../state/search-app-state';
 import {GeneratedAnswerSection} from '../../state/state-sections';
 
-export function citationSourceSelector(
-  state: Partial<GeneratedAnswerSection>,
-  citationId: string
-) {
-  return state.generatedAnswer?.citations?.find(
-    (citation) => citation.id === citationId
-  );
-}
+export const citationSourceSelector = createSelector(
+  (state: Partial<GeneratedAnswerSection>, citationId: string) =>
+    state.generatedAnswer?.citations?.find(
+      (citation) => citation.id === citationId
+    ),
+  (citation) => citation
+);
```

Comment:
https://redux.js.org/usage/deriving-data-selectors#writing-memoized-selectors-with-reselect

You swapped things around: the first parameters, the input selector must be 'light-weight' and the output selector should do the heavy lifting

> "input selectors" should usually just extract and return values, and the "output selector" should do the transformation work

Reply:
I knew this...  of course I knew this, right :badpokerface: